src/beautifiers/rubocop.coffee

```diff
@@ -16,53 +18,50 @@ module.exports = class Rubocop extends Beautifier
       rubocop_path: true
   }
 
-  beautify: (text, language, options) ->
+  beautify: (text, language, options, context) ->
+    fullPath = context.filePath
+    [projectPath, relativePath] = atom.project.relativizePath(fullPath)
+
+    # Find the rubocop path
+    rubocopPath = "rubocop"
     @Promise.all([
       @which(options.rubocop_path) if options.rubocop_path
       @which('rubocop')
-    ]).then((paths) =>
+    ])
+    .then((paths) =>
       @debug('rubocop paths', paths)
-      _ = require 'lodash'
-      path = require 'path'
       # Get first valid, absolute path
-      rubocopPath = _.find(paths, (p) -> p and path.isAbsolute(p) )
+      rubocopPath = paths.find((p) -> p and path.isAbsolute(p))
       @verbose('rubocopPath', rubocopPath)
       @debug('rubocopPath', rubocopPath, paths)
+    )
 
-      configFile = path.join(atom.project.getPaths()[0], ".rubocop.yml")
-
-      fs = require 'fs'
+    # Find or generate a config file if non exists
+    if !@findFile(path.dirname(fullPath), ".rubocop.yml")?
+      yaml = require("yaml-front-matter")
+      config = {
+        "Style/IndentationWidth":
+          "Width": options.indent_size
+      }
+      tempConfig = @tempFile("rubocop-config", yaml.safeDump(config))
 
-      if fs.existsSync(configFile)
-        @debug("rubocop", config, fs.readFileSync(configFile, 'utf8'))
-      else
-        yaml = require("yaml-front-matter")
-        # Generate config file
-        config = {
-          "Style/IndentationWidth":
-            "Width": options.indent_size
-        }
+    rubocopArguments = [
+      "--auto-correct"
+      "--force-exclusion"
+      "--stdin", relativePath || temp.path({suffix: '.rb'})
```

Comment:
```coffeescript
 "--stdin", relativePath || temp.path({suffix: '.rb'})
```

When I read this I initial thought `"--stdin"` was associated with `relativePath`, however this is not true.

I recommend breaking this up into two lines, unless they are related:

```coffeescript
"--stdin"
relativePath || temp.path({suffix: '.rb'})
```

Reply:
It is associated with it, as I mentioned in the other comment, `--stdin` requires a parameter, so I believe the right thing is to leave it on that line.

Reply:
Understood. Leave it as is.